### PR TITLE
Stop and disconnect player mid-play

### PR DESCRIPTION
**Pull request by Halzorg**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -39,6 +39,16 @@ async def play(ctx, url):
...
 
+#Code to implement bot stop music command
+@bot.slash_command(description="Stops the music and disconnects the bot")
+async def stop(ctx):
+    if players.get(ctx.guild):
+        await players[ctx.guild].stop_and_disconnect()
+        del players[ctx.guild]  # Remove the player instance
+        await ctx.respond("Stopped and disconnected.")
+    else:
+        await ctx.respond("No active player found.")
```
Comment (Force-quit) [inline main.py:50]: Please pull from main into your branch to have the latest code. You should only have to call stop_and_disconnect() and the memory will be freed automatically.

File: player.py [hunk 2]
```diff
@@ -77,6 +77,14 @@ async def skip(self, ctx):
         else:
             await ctx.respond("There is no queue my friend")
 
+    #code to add a stop music and disconnect command
+    async def stop_and_disconnect(self):
+        if self._voice_client.is_playing():
+            self._voice_client.stop()
+        if self._voice_client.is_connected():
+            await self._voice_client.disconnect()
+        self._queue.clear()
```
Comment (Force-quit) [inline player.py:86]: You should have only line 85 in here. If this object exists, it's connected to a voice chat. You can leave line 84 if you want, but it shouldn't be necessary. After the disconnect this object is deleted automatically, therefore you can remove line 86.

**Review by Force-quit — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (Force-quit) [reply main.py:50]: The feedback is good though you can keep those.

**Comment by Halzorg** [comment]
HI I just pulled the up to date code and committed my changes according to your comments.

**Comment by Force-quit** [comment]
All good. You can merge the pull request and delete your branch 

image

**Comment by Halzorg** [comment]
I don't have the option to merge "Only those with [write access] to this repository can merge pull requests.

**Comment by Force-quit** [comment]
Oh ok I'll merge it then. Thanks for contributing to our project!

**Pull request merged**